### ISSUE TITLE
MedicationRequest.doNotPerform is not included in Highlighting and Coverage Calculations

### ISSUE THREAD

**Issue by RohitKandimalla** [projecttacoma/fqm-execution#304]
### Summary
MADiE team received a helpdesk ticket where user is having issues achieving 100% for their measure that includes MedicationRequest.doNotPerform and the highlighting is also showing that it is not covered.

#### External Tracking Ticket
[BONNIEMAT-1835](https://oncprojectracking.healthit.gov/support/browse/BONNIEMAT-1835)

### Expected Behavior
The attached test case should highlight `DementiaMedication.doNotPerform` in 
`"define Has Dementia Medications in Year Before or During Measurement Period"`. Also provided a List of all test cases which is expected to give 100% coverage.

### Inputs (e.g. Measure Bundle, Patient Bundle, CQL Library)
[MeasureBundle.zip](https://github.com/projecttacoma/fqm-execution/files/15078325/CMS165FHIR-v0.1.000-FHIR.json.zip)
[PatientBundle.txt](https://github.com/projecttacoma/fqm-execution/files/15078070/PatientBundle.txt)
[CMS165FHIR-v0.1.000-FHIR4-TestCases.zip](https://github.com/projecttacoma/fqm-execution/files/15078158/CMS165FHIR-v0.1.000-FHIR4-TestCases.zip)

### Relevant Calculation Options (e.g. Measurement Period, meta.profile Validation)
MP : 1/1/2025 - 12/31/2025

### cql-to-elm Version Used for Measure Logic Translation (if known)
3.3.2

**Comment by RohitKandimalla** [comment]
[Uploading CMS165FHIR-v0.1.000-FHIR.txt…]()

**Comment by hossenlopp** [comment]
Hi @RohitKandimalla,

Could you please upload the measure bundle to this ticket or the BONNIEMAT ticket? It looks like it didn't sucessfully upload here.

What version of fqm-execution is being used with this?

-Chris

**Comment by RohitKandimalla** [comment]
I have issues uploading some files on Github this morning, compressed it as a work around.
and here is the version we are currently using `fqm-execution v1.3.3`

**Comment by hossenlopp** [comment]
This is expected behavior of the highlighting. `doNotPerform` needs a "truthy" value to be considered covered. The user just needs to add a case similar to the `PatientBundle.txt` that you uploaded but with `doNotPerform` being set to `true` on the `MedicationRequest` resource to cover that clause. This also is a good test case for the situation that the medication was specifically requested to NOT be given.

**Comment by hossenlopp** [comment]
Clause highlighting and coverage is not based on just "executed or not" it is based on the logic being relevant, executed AND having a result that is either true, a non-empty list, or filled with a value. If we allowed for boolean `false` to count for clause coverage then logic such as `exists "Some Expression"` would result in the `exists` always being highlighted green/covered.

We reviewed the BONNIEMAT ticket linked to this and noticed it is using the QICore MedicationRequest profile which makes it so `doNotPerform` is fixed to `false`. The validity of the resources being executed will have been checked by the data provider before hitting this logic. So there is no need to actually check for `doNotPerform` to be `true` in the CQL as it will have been done in the profile validity checks.
